internal/promotion/runner/builtin/git_pusher.go

```diff
@@ -181,11 +199,25 @@ func (g *gitPushPusher) run(
 		return promotion.StepResult{Status: kargoapi.PromotionPhaseErrored},
 			fmt.Errorf("error getting last commit ID: %w", err)
 	}
+
+	// Using git provider to get commit url. Continuing even if provider
+	// cannot be implemented as the push will still have succeeded, we
+	// just can't construct the URL.
+	gitProvider, err := gitprovider.New(repoURL, nil)
+	commitURL := ""
+	if err != nil {
+		commitURL = "Unknown"
+		logger.Info("unable to construct commit URL from %s: %s", repoURL, err)
+	} else {
+		commitURL, _ = gitProvider.GetCommitURL(repoURL, commitID)
+	}
+
 	return promotion.StepResult{
 		Status: kargoapi.PromotionPhaseSucceeded,
 		Output: map[string]any{
-			stateKeyBranch: pushOpts.TargetBranch,
-			stateKeyCommit: commitID,
+			stateKeyBranch:    pushOpts.TargetBranch,
+			stateKeyCommit:    commitID,
+			stateKeyCommitURL: commitURL,
```

Comment:
Now that we're running into multiple outputs prefixed with "commit", it is probably better that we structure output like this:

```
branch: <branch>
commit:
  id: <sha>
  url: <commit url>
commitID: <sha> # Retained for backwards compatibility.
```